### ISSUE TITLE
Unable to override -std:gnu++14

### ISSUE THREAD

**Issue by ibsusu** [nim-lang/Nim#13933]
Function `echo` outputs the wrong string.

### hello.nim
```nim
{.passc: "-std=c++17".}
{.compile: "uWebSockets/src/App.h".}
echo "Hello World!"

# nim cpp --cincludes:"uWebSockets/uSockets/src/" hello.nim
# nim cpp --passC:-std=c++17 --cincludes:"uWebSockets/uSockets/src/" hello.nim
```

### Current Output
```
work/scratch/nim/uWebSockets/src/App.h: In lambda function:
work/scratch/nim/uWebSockets/src/App.h:154:18: error: ‘string_view’ is not a member of ‘std’
  154 |             std::string_view secWebSocketKey = req->getHeader("sec-websocket-key");
      |                  ^~~~~~~~~~~
work/scratch/nim/uWebSockets/src/App.h:154:18: note: ‘std::string_view’ is only available from C++17 onwards
work/scratch/nim/uWebSockets/src/App.h:155:17: error: ‘secWebSocketKey’ was not declared in this scope
```

### Expected Output
It should compile

### Possible Solution

Something to do with this?
https://github.com/nim-lang/Nim/blob/devel/compiler/extccomp.nim#L89

### Additional Information

This is my first ticket.  Should I just compile nim with that line and file changed?

```
$ nim -v
Nim Compiler Version 1.2.0 [Linux: amd64]
Compiled at 2020-04-03
Copyright (c) 2006-2020 by Andreas Rumpf

active boot switches: -d:release
```

**Comment by metagn** [comment]
Seems to be fixed in devel branch by #13798

**Comment by ibsusu** [comment]
Yup, closing.
